### DISCUSSION TITLE
Could Wix bundle ask if dependency is uninstalled?

### DISCUSSION THREAD

**Discussion by RoyZhang2022** [wixtoolset/issues#6913] [Questions] [answered]
Hi,

Currently our msi has a dependency for Microsoft VC redistributed package on Windows. So I use Wix bundle to install VC_redist.x64.exe first, then install our own msi. The installation is successful. For the uninstallation our application could be removed but the dependency VC redistributed package is still there. We understand it may be needed by other applications. So is there a way to pop up a window to confirm with user if user wants to uninstall it or not. So user could have a choice instead of leaving the dependency there untouched. I searched bundle, ExePackage and Chain. I did not see such an option. Does someone know how to achieve it or it is not doable? Thanks in advance!

```
  9     <BootstrapperApplicationRef Id="WixStandardBootstrapperApplication.RtfLicense">
 10           <bal:WixStandardBootstrapperApplication
 11             LicenseFile="license.rtf"
 13             ShowVersion="yes"
 14             />
 15     </BootstrapperApplicationRef>
 16
 17     <util:FileSearch Path="[WindowsFolder]System32\vcruntime140.dll" Variable="VCDISTINSTALLED"/>
 18
 19     <Chain>
 20
 21       <ExePackage SourceFile="VC_redist.x64.exe"
 22                   DetectCondition="VCDISTINSTALLED"
 23                   InstallCommand="/q /ACTION=Install"
 24                   RepairCommand="/q ACTION=Repair /hideconsole"
 25                   />
 26
 27       <MsiPackage SourceFile="our.msi" />
 28
 29     </Chain>
```

Regards,
Roy

**Comment by robmen** [comment, marked as answer]
Yes, but the `WixStandardBootstrapperApplication` does not support that scenario. So you'd need to develop your own bootstrapper application (BA) to implement that feature (plus all of the standard features required by a BA). It's a good but of work.

**Reply by RoyZhang2022** [reply]
Thanks for your reply! Is there a guide for development of bootstrapper application?

**Reply by robmen** [reply]
I don't know that any one has written a complete guide. There are existing boostrapper applications in the WiX Toolset source code that you can read.
